beacon-chain/sync/subscriber.go

```diff
@@ -193,26 +206,48 @@ func (s *Service) wrapAndReportValidation(topic string, v pubsub.ValidatorEx) (s
 		}
 		// Ignore any messages received before chainstart.
 		if s.chainStarted.IsNotSet() {
-			messageFailedValidationCounter.WithLabelValues(topic).Inc()
+			messageIgnoredValidationCounter.WithLabelValues(topic).Inc()
+			return pubsub.ValidationIgnore
+		}
+		retDigest, err := p2p.ExtractGossipDigest(topic)
+		if err != nil {
+			log.WithField("topic", topic).Errorf("Invalid topic format of pubsub topic: %v", err)
+			return pubsub.ValidationIgnore
+		}
+		currDigest, err := s.currentForkDigest()
+		if err != nil {
+			log.WithField("topic", topic).Errorf("Unable to retrieve fork data: %v", err)
+			return pubsub.ValidationIgnore
+		}
+		if currDigest != retDigest {
+			log.WithField("topic", topic).Debugf("Received message from outdated fork digest %#x", retDigest)
 			return pubsub.ValidationIgnore
 		}
 		b := v(ctx, pid, msg)
 		if b == pubsub.ValidationReject {
 			messageFailedValidationCounter.WithLabelValues(topic).Inc()
 		}
+		if b == pubsub.ValidationIgnore {
+			messageIgnoredValidationCounter.WithLabelValues(topic).Inc()
+		}
 		return b
 	}
 }
 
 // subscribe to a static subnet  with the given topic and index.A given validator and subscription handler is
 // used to handle messages from the subnet. The base protobuf message is used to initialize new messages for decoding.
-func (s *Service) subscribeStaticWithSubnets(topic string, validator pubsub.ValidatorEx, handle subHandler) {
-	base := p2p.GossipTopicMappings(topic, 0)
+func (s *Service) subscribeStaticWithSubnets(topic string, validator pubsub.ValidatorEx, handle subHandler, digest [4]byte) {
+	genRoot := s.cfg.Chain.GenesisValidatorRoot()
+	_, e, err := p2putils.RetrieveForkDataFromDigest(digest, genRoot[:])
+	if err != nil {
+		panic(err)
```

Comment:
panic? Can't we change the signature to return an error?

Reply:
Same reason as mentioned above